src/sequelize.js

```diff
@@ -1271,6 +1271,10 @@ Use Sequelize#query if you wish to use replacements.`);
       }
     }
 
+    if (attribute.allowNull !== false && !attribute.defaultValue) {
```

Comment:
`defaultValue` could be 0, or `''`, a better check would be:

```suggestion
    if (attribute.allowNull !== false && attribute.defaultValue === undefined) {
```

It's also a good opportunity to normalize `attribute.allowNull` just before this so the test can be reduced to

```suggestion
    if (attribute.allowNull && attribute.defaultValue === undefined) {
```